### PR TITLE
Pythonized RDF Define

### PR DESCRIPTION
**Pull request by Pawan-Johnson**
Implemented pythonization to RDF Define.
Define can now directly accept a python callable.
# This Pull request:

## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

**Comment by vepadulano** [comment]
@phsft-bot build

**Comment by vepadulano** [comment]
@phsft-bot build

File: bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py [hunk 1]
```diff
@@ -24,7 +22,7 @@ class FunctionJitter:
     Attributes
     ----------
     rdf: RDataFrame object on which the function's column arguments are in
-    col_names: The list of columns in rdf
+    col_names: The list of columns in rdf
```
Comment (etejedor) [inline bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py:25]: Extra space (also in other lines below, please check).

Comment (Pawan-Johnson) [reply bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py]: I have fixed the spaces. Hopefully. 
Not sure where the auto-formatting was triggered from. Something with the editor.

File: bindings/pyroot/pythonizations/test/rdf_define_pyz.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import unittest
```
Comment (vepadulano) [inline bindings/pyroot/pythonizations/test/rdf_define_pyz.py:1]: Also, please add at least one test where the input types of the python callable are columns that hold vectors and they are used as numpy arrays inside the python callable

**Comment by etejedor** [comment]
Reviewed together with @vepadulano, please see his comments. Congrats @Pawan-Johnson for your work and this nice and clean solution!

File: bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py [hunk 3]
```diff
@@ -315,3 +293,58 @@ def x_more_than_y(x):
...
+
+    if isinstance(callable_or_str, str): # If string argument is passed. Invoke the Original Define.
+        return rdf._OriginalDefine(col_name, callable_or_str)
+
+    if not callable(callable_or_str): # The 2st argument is either a string or a python callable.
+        raise TypeError(f"The second argument of a Define operation should be a callable. {type(callable_or_str).__name__} object is not callable.")
+    
+    if not isinstance(cols, list):        
+        raise TypeError(f"Define takes a column list as third arguments but {type(cols).__name__} was given.")
+
```
[the remark this answers is not in the document]
Comment (Pawan-Johnson) [reply bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py:337]: This and the previous comment on cols being any iterable. 
They are not specific to the Defines. The same need to be done for the Filters and DefinePerSample
So, I'll put them in a separate PR, if that makes sense.
Otherwise, we would have modifications to _PyFilter in a PR that adds _PyDefine functionality.

File: bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py [hunk 4]
```diff
@@ -286,10 +275,10 @@ def x_more_than_y(x):
     import libcppyy
      # Implies a cppyy proxy of a function was passed.
     if type(callable_or_str) == libcppyy.CPPOverload:
-        return rdf._OriginalFilter(callable_or_str, *args)
+        return rdf._OriginalFilter(func, *args)
      # Second condition is a Python proxy to an std::function
     if (isinstance(getattr(callable_or_str, 'target_type', None), libcppyy.CPPOverload)):
-        return rdf._OriginalFilter(callable_or_str, *args)
+        return rdf._OriginalFilter(func, *args)
```
Comment (vepadulano) [inline bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py:281]: These changes do not belong to this PR

File: bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py [hunk 5]
```diff
@@ -315,3 +304,58 @@ def x_more_than_y(x):
...
+    
+    func = callable_or_str
+    # Check if it is a c++ callable.
+    import libcppyy
+     # Implies a cppyy proxy of a function was passed.
+    if type(callable_or_str) == libcppyy.CPPOverload:
+        return rdf._OriginalDefine(col_name, func, cols)
+     # Second condition is a Python proxy to an std::function
+    if (isinstance(getattr(callable_or_str, 'target_type', None), libcppyy.CPPOverload)):
+        return rdf._OriginalDefine(callable_or_str, func, cols)
```
Comment (vepadulano) [inline bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py:357]: Why is `callable_or_str` being passed as first argument here?

Comment (vepadulano) [inline bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py:357]: Most probably these two checks can be merged in one line

**Review by vepadulano — CHANGES_REQUESTED** [review]
Getting closer, see further comments

**Comment by vepadulano** [comment]
@phsft-bot build

**Review by vepadulano — APPROVED** [review]
Thank you Pawan!
